src/components/FileUploadModal.tsx

```diff
@@ -0,0 +1,225 @@
+import {
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogOverlay,
+  DialogPortal,
+  DialogTitle,
+  DialogTrigger,
+} from "@radix-ui/react-dialog";
+import { Accept, useDropzone } from "react-dropzone";
+import { useState } from "react";
+
+// Icon Imports
+import submitIcon from "@/assets/icons/submitIcon.svg";
+import crossIcon from "@/assets/icons/xIconPrimary.svg";
+import alertIcon from "@/assets/icons/alert.svg";
+import fileLoadIcon from "@/assets/icons/fileLoadSuccessIcon.svg";
+import uploadGreenIcon from "@/assets/icons/uploadGreen.svg";
+import { lookup } from "mime-types";
+
+type FileUploadModalProps = {
+  children: React.ReactNode;
+  onUpload: (files: File[]) => void;
+  acceptedFileExtensions: string[];
+  maxFileSize: number; // MB
+};
+
+type UploadState = "success" | "fail" | "none";
+
+export default function FileUploadModal({
+  children,
+  onUpload,
+  acceptedFileExtensions,
+  maxFileSize,
+}: FileUploadModalProps) {
+  let [stagedFiles, setStagedFiles] = useState<File[]>([]);
```

Comment:
better way to handle these might be to have a files state of objects that contain the file and any relevant info about their upload  

```suggestion
  let [files, setFiles] = useState<{ file: File; status: 'pending' | 'success' | 'failure' }>([]);
```